### ISSUE TITLE
[v4] `Attempted to assign to readonly property` error

### ISSUE THREAD

**Issue by ta5een** [gorhom/react-native-scroll-bottom-sheet#650]
# Bug

I created a simple button that, when clicked, expands a `BottomSheet` like so:

```tsx
// Code to open bottom sheet, nothing special...
const handleExpandBottomSheet = () => {
  bottomSheetRef.current?.expand();
};

// Later in the code...
return (
  <View>
    <Button title="Expand Bottom Sheet" onPress={handleExpandBottomSheet} />
    <BottomSheet
      enablePanDownToClose
      index={-1}
      ref={bottomSheetRef}
      snapPoints={bottomSheetSnapPoints}
      onChange={handleBottomSheetChanges}>
      {/* ... Just a simple View here ... */}
    </BottomSheet>;
  </View>
);
```

As expected, when I click the button, the bottom sheet opens up. However, when I navigate back (pop off the current screen) and then come back to it, I immediately get the error: `Attempted to assign to readonly property`. This error occurs before the screen renders. It points to this specific location in the source (Line 33):

https://github.com/gorhom/react-native-bottom-sheet/blob/f03b05bbc39bf62f7d97422e717f2998f2e1fada/src/utilities/animate.ts#L29-L35

This issue _only_ occurs on iOS. It happens on both a physical device and the Simulator. It does _NOT_ happen when I use version 3 (specifically `3.6.6`). I am using React Navigation 6 to navigate through the screens, if that's relevant at all.

When I tried to replicate this error on a fresh React Native project, I cannot get the error to appear. I believe this error is appearing due to me setting up something incorrectly.

<details>
  <summary>Babel configuration</summary>

```js
module.exports = {
  presets: [
    'module:metro-react-native-babel-preset',
    ['@babel/preset-env', { targets: { node: 'current' } }],
    '@babel/preset-typescript',
  ],
  plugins: [
    ['react-native-reanimated/plugin'],
    [
      require.resolve('babel-plugin-module-resolver'),
      {
        root: ['./src'],
        extensions: ['.ios.js', '.android.js', '.js', '.ts', '.tsx', '.json'],
        alias: {
          src: './src',
        },
      },
    ],
    ['@babel/plugin-transform-flow-strip-types'],
    ['@babel/plugin-proposal-decorators', { legacy: true }],
    ['@babel/plugin-proposal-class-properties', { loose: true }],
    ['@babel/plugin-proposal-private-property-in-object', { loose: true }],
  ],
};
```

</details>

<details>
  <summary>Full error message</summary>

```
TypeError: Attempted to assign to readonly property

This error is located at:
  in BottomSheet (created by BottomSheet)
  in RCTSafeAreaView (at SafeAreaView.js:51)
  in SafeAreaView (at SettingsNavigator.tsx:29)
  in MainSettingsScreen (at SceneView.tsx:126)
  in StaticContainer
  in EnsureSingleNavigator (at SceneView.tsx:118)
  in SceneView (at useDescriptors.tsx:209)
  in RCTView (at View.js:34)
  in View (at CardContainer.tsx:280)
  in RCTView (at View.js:34)
  in View (at CardContainer.tsx:278)
  in RCTView (at View.js:34)
  in View (at CardSheet.tsx:33)
  in CardSheet (at Card.tsx:556)
  in RCTView (at View.js:34)
  in View (at createAnimatedComponent.js:217)
  in AnimatedComponent (at createAnimatedComponent.js:278)
  in AnimatedComponentWrapper (at Card.tsx:535)
  in PanGestureHandler (at GestureHandlerNative.tsx:14)
  in PanGestureHandler (at Card.tsx:529)
  in RCTView (at View.js:34)
  in View (at createAnimatedComponent.js:217)
  in AnimatedComponent (at createAnimatedComponent.js:278)
  in AnimatedComponentWrapper (at Card.tsx:525)
  in RCTView (at View.js:34)
  in View (at Card.tsx:519)
  in Card (at CardContainer.tsx:218)
  in CardContainer (at CardStack.tsx:649)
  in RNSScreen (at createAnimatedComponent.js:217)
  in AnimatedComponent (at createAnimatedComponent.js:278)
  in AnimatedComponentWrapper (at src/index.native.tsx:169)
  in Screen (at Screens.tsx:37)
  in MaybeScreen (at CardStack.tsx:642)
  in RNSScreenContainer (at src/index.native.tsx:236)
  in ScreenContainer (at Screens.tsx:20)
  in MaybeScreenContainer (at CardStack.tsx:561)
  in RCTView (at View.js:34)
  in View (at Background.tsx:13)
  in Background (at CardStack.tsx:559)
  in CardStack (at StackView.tsx:437)
  in RNCSafeAreaProvider (at SafeAreaContext.tsx:76)
  in SafeAreaProvider (at SafeAreaProviderCompat.tsx:46)
  in SafeAreaProviderCompat (at StackView.tsx:430)
  in RCTView (at View.js:34)
  in View (at StackView.tsx:429)
  in StackView (at createStackNavigator.tsx:118)
  in Unknown (at createStackNavigator.tsx:117)
  in StackNavigator (at RootNavigator.tsx:15)
  in MainNavigator (at AuthGate.tsx:171)
  in AuthGate (at App.tsx:89)
  in EnsureSingleNavigator (at BaseNavigationContainer.tsx:429)
  in BaseNavigationContainer (at NavigationContainer.tsx:132)
  in ThemeProvider (at NavigationContainer.tsx:131)
  in NavigationContainerInner (at App.tsx:65)
  in PersistGate (at App.tsx:52)
  in Provider (at App.tsx:51)
  in App (at renderApplication.js:47)
  in RCTView (at View.js:34)
  in View (at AppContainer.js:107)
  in RCTView (at View.js:34)
  in View (at AppContainer.js:134)
  in AppContainer (at renderApplication.js:40)
```

</details>

![Error Screenshot copy](https://user-images.githubusercontent.com/37362870/133965587-e4170fbc-2413-4d60-b24a-1f7a9daafc32.png)

## Environment info

| Library                      | Version |
| ---------------------------- | ------- |
| @gorhom/bottom-sheet         | 4.0.3   |
| react-native                 | 0.64.2  |
| react-native-reanimated      | 2.2.0   |
| react-native-gesture-handler | 1.10.3  |

## Steps To Reproduce

1. From `ScreenA`, navigate to `ScreenB` via React Navigation
2. In `ScreenB`, click on the button that opens a `BottomSheet`
3. Navigate back (pop off the screen) to `ScreenA`. No need to close `BottomSheet` as error happens regardless
4. Navigate to `ScreenB` again
5. Observe the error (occurs before `ScreenB` completely renders)

Describe what you expected to happen:

1. No error appears when I navigate to `ScreenB`
2. When I click on the same button, the `BottomSheet` expands

## Reproducible sample code

Issue cannot be reproduced on a fresh project.

**Comment by Akumzy** [comment]
Came here to report as well

**Comment by SCasarotto** [comment]
Am also seeing this issue only in release scheme or deployed app.

EDIT: Some more information. I realized that I am seeing this error on v2. I can make another issue for this if thats more appropriate. I first saw this error on 2.3.0 and just upgraded to 2.4.0 and am seeing the same issue.

**Comment by gorhom** [comment]
this should be fixed on `4.1.0`
